### DISCUSSION TITLE
Why does `Any` make variable implicit type alias?

### DISCUSSION THREAD

**Discussion by Molkree** [microsoft/pyright#4460] [General]
I've noticed that Pyright considers variables that don't have explicit type annotation to be type alias when being assigned `Any` value, for example:
```py
from typing import Any

foo: Any = 111
bar = foo
```
Here `bar` has this tooltip from Pyright: `(type alias) bar: Any`. Which in turn results in this code being accepted:
```py
baz: bar = "sss"
```
mypy raises this error: `Variable "test.bar" is not valid as a type  [valid-type]`.

Before Pyright [1.1.117](https://github.com/microsoft/pyright/releases/tag/1.1.117) `bar` was also a variable. That release has this change:
> Behavior Change: Changed the logic that determines whether a variable assignment is an implicit type alias definition. If there is an explicit type annotation (other than the use of the PEP 612 TypeAlias), it is no longer considered a type alias. This is consistent with the rules mypy uses.

It looks like we were supposed to get fewer type aliases and more just variables but this specific case was instead changed from variable to type alias? As evident by the error this is not consistent with mypy (or maybe they changed logic since 2021).

Was this intentional? If yes, why change from variable to implicit type alias?

P.S. I haven't encountered any problems from this behavior, just a confusing tooltip saying that my variable was type alias in this code:
```py
from email.message import EmailMessage

msg = EmailMessage()
msg["Subject"] = "Subject line"
content = msg["Subject"] + ":\nMore content"
# above is Any + str, which results in (type alias content: Any)
```

**Comment by erictraut** [comment]
There are no defined rules for determining whether an unannotated variable should be treated as a type alias. Pyright uses a bunch of heuristics. Changing these heuristics tends to be disruptive, so unless there's a compelling reason to touch that code, I leave it as is.
